src/audio/base_fw.c

```diff
@@ -419,6 +421,44 @@ static int basefw_power_state_info_get(uint32_t *data_offset, char *data)
 	return 0;
 }
 
+static int basefw_libraries_info_get(uint32_t *data_offset, char *data)
+{
+	struct ipc4_libraries_info *const libs_info = (struct ipc4_libraries_info *)data;
+	struct sof_man_fw_desc *desc = NULL;
+
+	libs_info->library_count = 0;
+
+	for (int lib_id = 0; lib_id < LIB_MANAGER_MAX_LIBS; ++lib_id) {
+		if (lib_id == 0)
+			desc = (struct sof_man_fw_desc *)IMR_BOOT_LDR_MANIFEST_BASE;
+		else
+			desc = lib_manager_get_library_module_desc(lib_id);
+
+		if (!desc)
+			continue;
+
+		libs_info->libraries[libs_info->library_count].id = lib_id;
+		memcpy_s(libs_info->libraries[libs_info->library_count].name,
+			 SOF_MAN_FW_HDR_FW_NAME_LEN, desc->header.name, sizeof(desc->header.name));
+		libs_info->libraries[libs_info->library_count].major_version =
+			desc->header.major_version;
+		libs_info->libraries[libs_info->library_count].minor_version =
+			desc->header.minor_version;
+		libs_info->libraries[libs_info->library_count].hotfix_version =
+			desc->header.hotfix_version;
+		libs_info->libraries[libs_info->library_count].build_version =
+			desc->header.build_version;
+		libs_info->libraries[libs_info->library_count].num_module_entries =
+			desc->header.num_module_entries;
+
+		libs_info->library_count++;
+	}
+
+	*data_offset =
+		sizeof(uint32_t) + libs_info->library_count * sizeof(struct ipc4_library_props);
```

Comment:
I'd use objects instead of types as a `sizeof()` argument

Reply:
Done, fixed